spring-ai-core/src/main/java/org/springframework/ai/vectorstore/filter/FilterExpressionBuilder.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.vectorstore.filter;
+
+import java.util.List;
+
+import org.springframework.ai.vectorstore.filter.Filter.ExpressionType;
+import org.springframework.ai.vectorstore.filter.Filter.Key;
+import org.springframework.ai.vectorstore.filter.Filter.Value;
+import org.springframework.ai.vectorstore.filter.converter.PrintFilterExpressionConverter;
+
+/**
+ * DSL builder for {@link Filter.Expression} instances. Here are some common examples:
+ *
+ * <pre>{@code
+ * var b = new FilterExpressionBuilder();
+ *
+ * // 1: country == "BG"
+ * var exp1 = b.eq("country", "BG");
+ *
+ * // 2: genre = "drama" AND year >= 2020
+ * var exp2 = b.and(b.eq("genre", "drama"), b.gte("year", 2020));
```

Comment:
I guess that should be b.in("generate", "drama")  ?

Reply:
there was a missing `=`. Should be correct now.